protocol-units/bridge/contracts/src/AtomicBridgeInitator.sol

```diff
@@ -0,0 +1,137 @@
+import {console} from "forge-std/Test.sol";
+import "./IAtomicBridgeInitiator.sol";
+import "./WETH/interfaces/IWETH10.sol";
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.6;
+
+contract AtomicBridgeInitiator is IAtomicBridgeInitiator {
+    struct BridgeTransfer {
+        uint256 amount;
+        address originator;
+        address recipient;
+        bytes32 hashLock;
+        uint timeLock;
+        bool exists;
+    }
+
+    mapping(bytes32 => BridgeTransfer) public bridgeTransfers;
+    mapping(bytes32 => BridgeTransfer) public completedBridgeTransfers;
+    IWETH10 public weth;
+
+    constructor(address _weth) {
+        weth = IWETH10(_weth);
+    }
+
+    function initiateBridgeTransfer(
+        uint256 _wethAmount,
+        address _originator, 
+        address _recipient, 
+        bytes32 _hashLock, 
+        uint _timeLock
+    ) external payable override returns (bytes32 _bridgeTransferId) {
+        console.log("initiateBridgeTransfer");
+        uint256 totalAmount = _wethAmount;
+
+        // If msg.value is greater than 0, convert ETH to WETH and add to total amount
+        if (msg.value > 0) {
+            weth.deposit{value: msg.value}();
+            //Transfer WETH to this contract, revert if transfer fails
+            require(weth.transfer(address(this), msg.value), "WETH transfer failed");
```

Comment:
Isn't this covered by the `weth.deposit` call already?

Reply:
actually, contract is making the deposit!

Reply:
Yes, this was so that it works with @cde8's traits and original design.